lib/tapioca/loaders/loader.rb

```diff
@@ -112,7 +110,14 @@ def load_engines_in_zeitwerk_mode
         autoloader = Zeitwerk::Loader.new
 
         engines.each do |engine|
-          engine.config.all_eager_load_paths.each do |path|
+          # https://github.com/rails/rails/commit/ebfca905db14020589c22e6937382e6f8f687664
+          eager_load_paths = if Rails::VERSION::MAJOR >= 7 && Rails::VERSION::MINOR >= 2
+            engine.config.all_eager_load_paths
+          else
+            engine.config.eager_load_paths
+          end
```

Comment:
I am not a huge fan of version checks if we don't have to, feature detection is always nicer. 

Can we instead check for the existence of the `all_eager_load_paths` method instead and encapsulate the lookup in a method?

```suggestion
          eager_load_paths = eager_load_paths(engine)
```
and

```ruby
def eager_load_paths(engine)
  config = engine.config

  (config.respond_to?(:all_eager_load_paths) && config.all_eager_load_paths) || config.eager_load_paths
end
```

Reply:
Agreed. I made the change.

However, we still need to duplicate the logic for the symbol loader, which I don't exactly love.

Reply:
Yes, that's not great, but I think the real fix is to extract an engine handler class to unify the logic. Not really a concern that needs to be addressed by this PR, though.